test/metabase/query_processor/middleware/add_dimension_projections_test.clj

```diff
@@ -479,3 +495,40 @@
                                                    &Q1.orders.product_id]}]
                           :fields [&Q1.orders.id &Q1.orders.product_id]
                           :limit  2})))))))))
+
+(deftest internal-remap-e2e-test
+  (mt/with-column-remappings [venues.category_id (values-of categories.name)]
+    (let [results (qp/process-query
+                   {:database (mt/id)
+                    :type     :query
+                    :query    {:source-table (mt/id :venues)
+                               :limit        2}})]
+      (is (= ["ID" "Name"  "Category ID" "Latitude" "Longitude" "Price" "Category ID [internal remap]"]
```

Comment:
This is mostly my own confusion, but where are the `[external remap]` and `[internal remap]` extras coming from? I looked at `with-column-remappings` but it's still really not clear.

Reply:
Yeah they come from `with-column-remappings`, if you look at `metabase.test.util/do-with-column-remappings` that's where those names are generated.